File: Course_01/C++ Language/Matrizes/diagonal_negativos.cpp
#include <iostream>

using namespace std;

int main()
{
    int n, i, j;

    cout << "Qual a ordem da matriz? ";
    cin >> n;

    int mat[n][n], conteNeg(0);

    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
    }

    cout << "DIAGONAL PRINCIPAL:" << endl;

    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            if (i == j) {
                cout << mat[i][j] << " ";
            }
            if (mat[i][j] < 0) {
                conteNeg++;
            }
        }
    }
    cout << endl;
    cout << "QUANTIDADE DE NEGATIVOS = " << conteNeg << endl;
    return 0;
}

File: Course_01/Python/Matrizes/diagonal_negativos.py
m = int(input("Qual a ordem da matriz? "))

matriz = [[0 for x in range(m)] for x in range(m)]

for i in range(0, m):
    for j in range(0, m):
        matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("Diagonal principal: ")

for i in range(0, m):
    print(f"{matriz[i][i]} ", end="")
print()
contagem = 0
for i in range(0, m):
    for j in range(0, m):
        if matriz[i][j] < 0:
            contagem += 1

print(f"Quantidade de negativos = {contagem}")